extensions/schema-compare/src/schemaCompareMainWindow.ts

```diff
@@ -1126,6 +1126,20 @@ export class SchemaCompareMainWindow {
 		if (ownerUri) {
 			endpointInfo = endpoint;
 			endpointInfo.ownerUri = ownerUri;
+		} else if (endpoint.endpointType === mssql.SchemaCompareEndpointType.Project) {
+			endpointInfo = {
+				endpointType: endpoint.endpointType,
+				packageFilePath: '',
+				serverDisplayName: '',
+				serverName: '',
+				databaseName: '',
+				ownerUri: '',
+				connectionDetails: undefined,
+				projectFilePath: endpoint.projectFilePath,
+				targetScripts: [],
```

Comment:
do the scripts get added after?

Reply:
Yes, just before the schema compare API is called (https://github.com/microsoft/azuredatastudio/blob/main/extensions/schema-compare/src/schemaCompareMainWindow.ts#L379) .